### DISCUSSION TITLE
Will L1 reorg cause L2 block hash change?

### DISCUSSION THREAD

**Discussion by bap2pecs** [ethereum-optimism/developers#459] [Application Development] [answered]
let's say we have a "safe" L2 block confirmed on L1. Then a L1 reorg happens so the batch transaction gets discards. Now the batch submitter will resubmit a batch containing the L2 block tx list. Will this lead to different L2 block hashes?

**Comment by bap2pecs** [comment]
@tynes is this the right place for asking questions like this?

**Reply by tynes** [reply]
Happy to discuss well though through questions about the protocol here. This isn't the place to ask implementation specific questions or ask for help about how to run the software. See https://github.com/ethereum-optimism/developers for that.

This question is relatively low effort as it doesn't go into the various nuance with the specific scenario you are describing. Happy to engage on high effort questions that take into account the specifics and are backed up with facts about the system. Its not scalable to respond with a long paragraph when the question is just a sentence or two.

**Comment by tynes** [comment, marked as answer]
> Will this lead to different L2 block hashes?

No it will not lead to different L2 block hashes, altho there is a bunch of nuance here

**Comment by bap2pecs** [comment]
closing as this topic should be posted in  https://github.com/ethereum-optimism/developers
